### PR TITLE
Add more PHP versions to the CI and bump actions

### PR DESCRIPTION
**Pull request by williamdes**

### CODE REVIEW

**Comment by williamdes** [comment]
I am not sure why there is no CI overview here, but here is the result: https://github.com/williamdes/Invoker/actions/runs/6120452457/job/16612348096

PHP >= 8.1
```
There was 1 error:

1) Invoker\Test\InvokerTest::should_invoke_callable_with_optional_parameter_before_required_parameter
Invoker\Exception\NotEnoughParametersException: Unable to invoke the callable because no value was given for parameter 1 ($baz)

/home/runner/work/Invoker/Invoker/src/Invoker.php:67
/home/runner/work/Invoker/Invoker/tests/InvokerTest.php:[22](https://github.com/williamdes/Invoker/actions/runs/6120452457/job/16612348312#step:6:23)3
```

**Comment by williamdes** [comment]
Seems like the fix for https://github.com/PHP-DI/PHP-DI/issues/562 does not work anymore.
Could you have a look @mnapoli ?
I am not sure to be able to fix it.
Checked the test and it looks okay

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 name: CI
 
+permissions:
+    contents: read
```
Comment (mnapoli) [inline .github/workflows/ci.yml]: Any reason for this change btw?

Comment (williamdes) [reply .github/workflows/ci.yml]: Good question, others seem to ask this too: https://github.com/zumba/json-serializer/pull/57#discussion_r1321159944


It's not required, but if missing the permissions given to the CI token are wider and more insecure if someone manages to steal the token

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Comment (mnapoli) [reply .github/workflows/ci.yml]: I'd rather not have this specific line here, I can tighten permissions via the GitHub settings.

**Comment by mnapoli** [comment]
CI should run again! With another commit we should have build statuses

File: .github/workflows/ci.yml (hunk 1 above)
Comment (williamdes) [reply .github/workflows/ci.yml]: Okay, I will revert this

**Comment by williamdes** [comment]
> CI should run again! With another commit we should have build statuses

It did run, now we have the failure as expected 🎉 
Can this be merged even with the CI failure ?

**Pull request merged**